### DISCUSSION TITLE
Behringer X-Touch jamulus mixer control

### DISCUSSION THREAD

**Discussion by iamthechrisb** [jamulussoftware/jamulus#2240] [Technical Support & Help]
been taking a look at getting my Behringer X-Touch controlling fader / solo / mute / pan in jamulus 

Looking over https://github.com/jamulussoftware/jamulus/issues/95 I see some of the progress with --ctrlmidich,

originally I see that @corrados was using this with X-touch, then I see contributors using the Korg NanoKontrol2, It looks like the basics for the korg are reflected as the example in https://jamulus.io/wiki/Tips-Tricks-More. Is that correct? if so, is there anyone that has some specifics on the X-TOUCH -- the --ctrlmidich command line -- AND indicate what mode they are running the X-TOUCH in (e.g HUI / MC..etc)

Thank You -- hoping for some real faders and buttons :-)
ChrisB

**Comment by pcar75** [comment]
Have you looked at #2220 ?
Not specifically X-Touch but very helpful.

**Reply by iamthechrisb** [reply]
Very helpful -- thank you, did not realize #2220 was out there when I posted.. It also makes me wonder why there are no tools/map for the full x-touch vs the other models they have out there - but that's another topic..

**Comment by henkdegroot** [comment]
In the code I found the following text:

    // parse the server info string according to definition: there is
    // the legacy definition with just one or two numbers that only
    // provides a definition for the controller offset of the level
    // controllers (default 70 for the sake of Behringer X-Touch)
    // [MIDI channel];[offset for level]

So the the X-Touch, I would say the following command line option should work:

`--ctrlmidich "1;70"`

Though I am not sure about the midi channel to use for this device. You might just use 0, which should then make Jamulus "listen" to all channels.

I don't know which protocol mode you should use on the X-Touch. Looks like MC would be the one I would try first.

**Reply by iamthechrisb** [reply]
Thanks for the reply! I do a lot with audio and play guitar, but some of the midi stuff I've just kind of put up with over the years.. I think its probably 'Ctrl' mode -- that looks like the kind of midi jamulus is talking..and is the one that shows up with offset 70. . I'm not well versed in the protocols, but I believe MC uses different, finer, values for control -- It looks somewhat like MIDI but needs to be talking to something that can interpret MC. 

Maybe that's something we can hope for -- that someone is interested in working on? I think adding MC support would likely be a huge leap here. Looking forward -- OSC would be a good fit as well. I think base support for midi is working -- but the current standard for control surface looks to be MC -- and possible the future is OSC.  I'm sure there is also a further look at how channels are placed and named in the jamulus session, possible ways to set some params,  for this to be usable/repeatable/reliable in any situation where people are coming and going. 

Are there any special licensing considerations to incorporate MC and OSC? does anyone know?

**Reply by pljones** [reply]
Given the things being controlled only have granularity of integer percentage - at best - I wouldn't worry about "better" than MIDI for controlling Jamulus.

OSC is strings across a TCP/IP connection - open licence, as far as I know.

I'm not familiar with MC.  Got any references?

**Comment by pcar75** [comment]
Bonjour, this might be helpful too.
https://sites.google.com/view/mackiecontroluniversaldiyguide/home

**Reply by pljones** [reply]
Ah, that's MC?  In that case, it's likely not to be open. :)

**Reply by pcar75** [reply]
_Ref.: Mackie Control Universal (2003) a.k.a.  MCU = MC + Logic Control + HUI ; https://en.wikipedia.org/wiki/Human_User_Interface_Protocol_

_Ref.: http://www.midibox.org/dokuwiki/mc_protocol_mappings_

**Reply by iamthechrisb** [reply]
hard to tell, looks like there are many that are doing things with it -- I'll do some digging.
